include/boost/url/detail/replacement_field_rule.hpp

```diff
@@ -0,0 +1,74 @@
+//
+// Copyright (c) 2022 Alan de Freitas (alandefreitas@gmail.com)
+//
+// Distributed under the Boost Software License, Version 1.0. (See accompanying
+// file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+//
+// Official repository: https://github.com/boostorg/url
+//
+
+#ifndef BOOST_URL_DETAIL_REPLACEMENT_FIELD_RULE_HPP
+#define BOOST_URL_DETAIL_REPLACEMENT_FIELD_RULE_HPP
+
+#include <boost/url/error.hpp>
+#include <boost/url/string_view.hpp>
+#include <boost/url/grammar/variant_rule.hpp>
+#include <boost/url/grammar/unsigned_rule.hpp>
+
+namespace boost {
+namespace urls {
+namespace detail {
+
+// replacement_field ::=  "{" [arg_id] [":" format_spec "}"
+struct replacement_field_rule_t
```

Comment:
Do these need to be in this file or could they just go to the .ipp where they are used?

Reply:
It's kind of both. Only .ipp files include it. But it's in a separate header because many .ipp files use these rules. The router will use them too.